marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx

```diff
@@ -121,6 +121,69 @@ describe('TilesView component', () => {
         expect(document.querySelector('#flows-container')).toBeInTheDocument();
         expect(getByText('Create Flow')).toBeInTheDocument();
         expect(getByText('testFlow')).toBeInTheDocument();
+        // delete should work
+        fireEvent.click(getByTestId('deleteFlow-0'));
+        // testing that confirmation modal appears
+        expect(queryByText('Yes')).toBeInTheDocument();
+        fireEvent.click(getByText('No'));
+
+        // test description
+        fireEvent.click(getByText('testFlow'));
+        expect(getByText('Save')).not.toBeDisabled();
+        fireEvent.click(getByText('Cancel'));
+        // test run
+        fireEvent.click(getByLabelText('icon: right'));
+        expect(getByTestId('runStep-1')).toBeInTheDocument();
+    });
+
+    test('Verify run tile cannot edit with only readStep authority', async () => {
+        const authorityService = new AuthoritiesService();
+        authorityService.setAuthorities(['readStep']);
+        const {getByLabelText, getByText, queryByText, getByTestId} = render(<AuthoritiesContext.Provider value={authorityService}><TilesView/></AuthoritiesContext.Provider>);
+
+
+        // Run tile not shown initially
+        expect(queryByText("icon-run")).not.toBeInTheDocument();
+        expect(queryByText("title-run")).not.toBeInTheDocument();
+
+        fireEvent.click(getByLabelText("tool-run"));
+
+        // Run tile shown with entityTypes after click
+        expect(await(waitForElement(() => getByLabelText("icon-run")))).toBeInTheDocument();
+        expect(getByLabelText("title-run")).toBeInTheDocument();
+        expect(document.querySelector('#flows-container')).toBeInTheDocument();
+        expect(getByText('testFlow')).toBeInTheDocument();
+        // create flow shouldn't be provided
+        expect(queryByText('Create Flow')).toBeDisabled();
+        // delete should not work
+        fireEvent.click(getByTestId('deleteFlow-0'));
+        // testing that confirmation modal didn't appear
+        expect(queryByText('Yes')).not.toBeInTheDocument();
+        // test description
+        fireEvent.click(getByText('testFlow'));
+        expect(queryByText('Save')).not.toBeInTheDocument();
+        fireEvent.click(getByText('Close'));
+
+        // test run
+        fireEvent.click(getByLabelText('icon: right'));
+        expect(getByTestId('runStepDisabled-1')).toBeInTheDocument();
+
```

Comment:
Can we pls add some more assertions per test plan.

- in TilesView.test to check step runner is not able to delete a step
- in Load.test and Curate.test to check step-runner is not able to add steps to a flow 
- in Load.test and Curate.test same tests as the ones for ingestion/mapping reader authority but using readStep authority. Additionally these tests will have checks to verify that readStep auth can see Match/Merge and Custom tabs. So corresponding configs needs to be added in runConfig.

Reply:
The user with the readStep authority alone won't have access to the Load and Curate tiles.

Reply:
Right, how about runStep authority? HC roles wiki mentions that hub-central-step-runner will have following roles assigned: 
data-hub-flow-reader, data-hub-ingestion-reader, data-hub-mapping-reader, data-hub-match-merge-reader and data-hub-custom-reader

So, above roles enable a step-runner to access steps so they can be run in HC, but not view them in HC?

Reply:
Correct. They will be able to read the documents in the database, but won't be able to access the tiles or middle-tier CRUD APIs.